cns/imds/client.go

```diff
@@ -102,10 +104,50 @@ func (c *Client) GetVMUniqueID(ctx context.Context) (string, error) {
 	return vmUniqueID, nil
 }
 
-func (c *Client) getInstanceComputeMetadata(ctx context.Context) (map[string]any, error) {
+func (c *Client) GetNCVersionsFromIMDS(ctx context.Context) (map[string]string, error) {
+	var networkData NetworkMetadata
+	err := retry.Do(func() error {
+		networkMetadata, err := c.getInstanceMetadata(ctx, imdsNetworkPath)
+		if err != nil {
+			return errors.Wrap(err, "error getting IMDS network metadata")
+		}
+
+		// Try to parse the network metadata as the expected structure
+		// Convert the map to JSON and back to properly unmarshal into struct
+		jsonData, err := json.Marshal(networkMetadata)
+		if err != nil {
+			return errors.Wrap(err, "error marshaling network metadata")
+		}
+
+		if err := json.Unmarshal(jsonData, &networkData); err != nil {
+			return errors.Wrap(err, "error unmarshaling network metadata")
+		}
+		return nil
+	}, retry.Context(ctx), retry.Attempts(c.config.retryAttempts), retry.DelayType(retry.BackOffDelay))
+	if err != nil {
+		return nil, err
+	}
+
+	ncVersions := make(map[string]string)
+	for _, iface := range networkData.Interface {
```

Comment:
When writing a client at this level, don't try to assume how it will be used by the caller. Faithful reproduction of the data in the language's types should be the goal. This makes the client more reusable, if it returns `([]NetworkInterface, error)`. Leave it up to the caller to transform the data as it fits the business logic needs.